### DISCUSSION TITLE
I really want to get any mention when I uploaded files.

### DISCUSSION THREAD

**Discussion by bigprongs** [rpldy/react-uploady#428] [Q&A] [answered]
@yoavniran first of all I'm very thank you for your previous help.
I saw many discussions about that.
but I think it is some necessary to get any mention about upload ended when multiple files uploading.
because some of developers wanted move on next step when all files uploaded.
please consider about that and please inform me your opinion or way. 
and I have some problems with ```useItemFinishListener``` it isn't working below code except ```useItemProgressListener```

```
const LogProgress = (item:any) =>{
  useItemProgressListener((item:any) => {
	  console.log(item)
  });
  useItemFinishListener((item:any) => {
            console.log(`item ${item.id} finished uploading, response was: `, item.uploadResponse, item.uploadStatus);  
        });
  
  return null;
}
```

**Comment by yoavniran** [comment, marked as answer]
Hey @elitesuper 
You can achieve what you need by tracking ITEM-START and ITEM-FINALIZE (not FINISH) events. 
you can do that simply with a counter that goes up on START and goes down on FINALIZE. When it gets to 0 it means no active uploads.

**Reply by bigprongs** [reply]
Thanks for your message! and then I also have a problem with a ```createMockSender```
I can't upload any files with ```createMockSender```.

Please check this code.
```
const LogProgress = () => {

	const [progress, setProgess] = useState(0);
	const progressData = useItemProgressListener();
  
	if (progressData && progressData.completed > progress) {
 	  console.log(progressData);
	  setProgess(() => progressData.completed);
	}

	return (<>
	progressData&&(<h1>{progress}</h1>)
	</>);
}
...
function CollectionUpload(props:any){
...
     <Uploady 
	     isSuccessfulCall={customIsSuccess} 
	     destination={{url: "http://localhost:5000/upload"}} 
	     {...nativeProps} 
	     enhancer={mockEnhancer}>
			    <UploadDropZone
			      className="collection-upload-field"
			      onDragOverClassName="drag-over"
			      htmlDirContentParams={{ recursive: true, withFullPath: true }}
			    >
			    <div className="upload-drop-zone">
				    <div className="upload-info">
				         <h5>Drag and drop or browse to choose your collection folder</h5>
	                                 <UploadButton text="Choose files..." className="voomio-btn btn-filled"/>
	                             </div>    
			    </div>    
			    </UploadDropZone>
	                     <LogProgress />
    </Uploady>
...
}
...
const mockEnhancer = (uploader:any) => {
	const mockSender = createMockSender({ delay: 1000 });
	uploader.update({ send: mockSender.send });
	return uploader;
};
```

**Reply by yoavniran** [reply]
What do you mean by "cant upload"? The purpose of the mock sender is to **NOT**  upload, but to only _mimic (mock)_ the upload process for testing/debug purposes (without a server).

**Reply by bigprongs** [reply]
yeah, you are right, so I've had a upload server already.
if I don't use mock, files successfully uploaded.
But if I use mock, it isn't working.

**Reply by bigprongs** [reply]
is that mock used by only debug purpose?

**Reply by bigprongs** [reply]
> Hey @elitesuper You can achieve what you need by tracking ITEM-START and ITEM-FINALIZE (not FINISH) events. you can do that simply with a counter that goes up on START and goes down on FINALIZE. When it gets to 0 it means no active uploads.

is it possible in multiple files?
how can I implement your opinion?
```
const LogProgress = () => {

	const [progress, setProgess] = useState(0);

	useItemStartListener((item) => {
		console.log(`useItemStartListener => item ${item.batchId}`);
	});
  
    useItemFinalizeListener((item) => {
        console.log(`useItemFinalizeListener => item ${item.batchId}`);
    });

	return null;
}
```

**Reply by yoavniran** [reply]
> is that mock used by only debug purpose?

Yes. Its only for your tests/debug

> Is it possible in multiple files?

of course. ITEM-START is called for every item when uploading begins
ITEM-FINALIZE is called for every upload that ends, successfully or not

**Reply by bigprongs** [reply]
Okay, Thanks!

**Comment by bigprongs** [comment]
Hi dear Thanks for your help, and I also have some problems again.
How do I view uploaded images in interface?
I'll wait your answer.
Kind regards.

**Reply by yoavniran** [reply]
https://react-uploady.org/docs/packages/rpldy-upload-preview/

https://codesandbox.io/s/preview-progress-react-uploady-pki4i

**Reply by bigprongs** [reply]
😒I thought There are more issues and problems than react-dropzone,
In react-dropzone, I can get all files(for instance total files number) and info before file upload then send server
but I think, In rpldy/react-uploady, I can't check total targetfile's number before send server.
so then I check total files number and  uploaded files number and failed files number with server response.
rpldy/react-uploady ignore server's activation.
this library only be used file uploaded I mean that can't do well with other interaction.
maybe I don't know about rpldy/react-uploady deeply.
